### PR TITLE
h2spec: handling PRIORITY frame that depend on itself

### PR DESCRIPTION
**Pull request by zizhong**
The issue reported by `h2spec` is
```Hypertext Transfer Protocol Version 2 (HTTP/2)
  5. Streams and Multiplexing
    5.3. Stream Priority
      5.3.1. Stream Dependencies
        × 2: Sends PRIORITY frame that depend on itself
          -> The endpoint MUST treat this as a stream error of type PROTOCOL_ERROR.
             Expected: GOAWAY Frame (Error Code: PROTOCOL_ERROR)
                       RST_STREAM Frame (Error Code: PROTOCOL_ERROR)
                       Connection closed
               Actual: Timeout
```
With this fix,
```
Hypertext Transfer Protocol Version 2 (HTTP/2)
  5. Streams and Multiplexing
    5.3. Stream Priority
      5.3.1. Stream Dependencies
             [send] SETTINGS Frame (length:6, flags:0x00, stream_id:0)
             [recv] SETTINGS Frame (length:12, flags:0x00, stream_id:0)
             [send] SETTINGS Frame (length:0, flags:0x01, stream_id:0)
             [recv] WINDOW_UPDATE Frame (length:4, flags:0x00, stream_id:0)
             [recv] SETTINGS Frame (length:0, flags:0x01, stream_id:0)
             [send] PRIORITY Frame (length:5, flags:0x00, stream_id:1)
             [recv] Connection closed
        ✔ 2: Sends PRIORITY frame that depend on itself

Finished in 0.0501 seconds
1 tests, 1 passed, 0 skipped, 0 failed
```

### CODE REVIEW

**Comment by masaori335** [comment]
[approve ci]

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/356/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/2038/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/469/

**Comment by atsci** [comment]
AU check *successful*! https://ci.trafficserver.apache.org/job/autest-github/340/

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/345/

**Comment by atsci** [comment]
Linux build *successful*! https://ci.trafficserver.apache.org/job/linux-github/1933/

**Comment by atsci** [comment]
clang-analyzer build *successful*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/602/

File: proxy/http2/Http2ConnectionState.cc [hunk 1]
```diff
@@ -387,6 +387,12 @@ rcv_priority_frame(Http2ConnectionState &cstate, const Http2Frame &frame)
                       "priority parse error");
   }
 
+  //  A stream cannot depend on itself.  An endpoint MUST treat this as a stream error of type PROTOCOL_ERROR.
+  if (stream_id == priority.stream_dependency) {
+    return Http2Error(Http2ErrorClass::HTTP2_ERROR_CLASS_CONNECTION, Http2ErrorCode::HTTP2_ERROR_PROTOCOL_ERROR,
```
Comment (maskit) [inline proxy/http2/Http2ConnectionState.cc]: The spec says treat it as a stream error, so please use `HTTP2_ERROR_CLASS_STREAM` instead.

Comment (zizhong) [inline proxy/http2/Http2ConnectionState.cc]: @maskit Thanks! Fixed.

**Comment by maskit** [comment]
Thanks.

**Comment by zwoop** [comment]
@maskit Do we want this in 7.1.0 ?

**Comment by maskit** [comment]
@zwoop I think we should. Masaori says it might cause memory leak.

**Comment by zwoop** [comment]
Cherry picked to 7.1.x, building and testing on docs now.

**Pull request merged**